core/input/input_event.cpp

```diff
@@ -887,36 +887,40 @@ bool InputEventJoypadMotion::is_pressed() const {
 	return Math::abs(axis_value) >= 0.5f;
 }
 
-bool InputEventJoypadMotion::action_match(const Ref<InputEvent> &p_event, bool *p_pressed, float *p_strength, float *p_raw_strength, float p_deadzone) const {
+bool InputEventJoypadMotion::action_match(const Ref<InputEvent> &p_event, bool p_exact_match, float p_deadzone, bool *r_pressed, float *r_strength, float *r_raw_strength) const {
 	Ref<InputEventJoypadMotion> jm = p_event;
 	if (jm.is_null()) {
 		return false;
 	}
 
-	bool match = (axis == jm->axis); // Matches even if not in the same direction, but returns a "not pressed" event.
+	// Matches even if not in the same direction, but returns a "not pressed" event.
+	bool match = axis == jm->axis;
+	if (p_exact_match) {
+		match &= (axis_value < 0) == (jm->axis_value < 0);
+	}
```

Comment:
I'm a bit concerned about the potential impact of this change on joypad motion events.

I guess let's merge and assess while testing whether this is working as expected.